### DISCUSSION TITLE
How to make tsup output declare module

### DISCUSSION THREAD

**Discussion by shuaixr** [egoist/tsup#771] [Q&A]
I use knex to manage mysql connections,knex uses `declare module 'knex/types/tables'` to enhance type checking([knex typescript doc](https://knexjs.org/guide/#typescript))

My type definition file is like this:
client.ts
```
import knex, { Knex } from "knex";
import knexConfig from "../knexfile";
//Init knex
export const client = knex(knexConfig);

declare module "knex/types/tables" {
//someting
}
```
tsc can correctly output type definition files containing `declare module 'knex/types/tables'`, but tsup can't

**Comment by thedanchez** [comment]
@shuaixr it's doable for sure. You just have to specify it as part of your `tsup.config.ts`

```ts
import { defineConfig } from 'tsup'

export default defineConfig((options) => ({
  name: "package-name,
  entry: ['src/index.tsx'],
  format: ["esm"],
  bundle: true,
  clean: true,
  dts: {
    footer: "declare module 'knex/types/tables';"
  },
}))
```
